libshortfin/src/shortfin/support/iree_helpers.h

```diff
@@ -105,41 +192,60 @@ class object_ptr {
     }
   }
   object_ptr(object_ptr &&other) : ptr(other.ptr) { other.ptr = nullptr; }
+  object_ptr &operator=(const object_ptr &other) = delete;
   object_ptr &operator=(object_ptr &&other) {
+    reset();
```

Comment:
Note that this was causing a reference leak, which just happened to be perfectly offsetting another ownership issue in the happy path.